src/PrestaShopBundle/EventListener/Context/Admin/CurrencyContextListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\EventListener\Context\Admin;
+
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Context\CurrencyContextBuilder;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+class CurrencyContextListener
+{
+    public function __construct(
+        private readonly CurrencyContextBuilder $currencyContextBuilder,
+        private readonly ConfigurationInterface $configuration,
+    ) {
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        $this->currencyContextBuilder->setCurrencyId((int) $this->configuration->get('PS_CURRENCY_DEFAULT'));
```

Comment:
Shall we add here a little protection for weird cases where 'PS_CURRENCY_DEFAULT' is `null`? A thrown exception or a fallback?

Reply:
It's a major database incosistency, an exception would be good.

Reply:
Indeed `PS_CURRENCY_DEFAULT` is one of the mandatory configurations always initialized by the install fixtures and always maintained even when it's modified If this configuration is absent the whole pricing system will basically explode both in BO and in FO

When that happens the problem won't be in this listener but much more global, so an additional check here would only make the code more complex without much value Besides the CurrencyBuilder service already checks the currency ID value when it build the CurrencyContext So it's checked regardless, just in another place